### PR TITLE
Bug fix/8076 throwing exception when trying to spy element on widgets window

### PR DESCRIPTION
**Pull request by ravirk91**
Bug 8090:Throwing Exception When trying to Refresh Controls from Window Explorer
Bug 8131:Trigger java script event failing with UI element action
Bug 8057:Need to filter POM specific target application on POM Edit Page

### CODE REVIEW

File: Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs [hunk 1]
```diff
@@ -291,6 +291,7 @@ public static void LoadUserMsgsPool()
             Reporter.UserMsgsPool.Add(eUserMsgKey.GoToUrlFailure, new UserMsg(eUserMsgType.ERROR, "Go To URL Error", "Failed to go to the URL: '{0}'." + Environment.NewLine + "Error Details: '{1}'.", eUserMsgOption.OK, eUserMsgSelection.None));
             Reporter.UserMsgsPool.Add(eUserMsgKey.HookLinkEventError, new UserMsg(eUserMsgType.ERROR, "Hook Link Event Error", "The link type is unknown.", eUserMsgOption.OK, eUserMsgSelection.None));
             Reporter.UserMsgsPool.Add(eUserMsgKey.AskToStartAgent, new UserMsg(eUserMsgType.WARN, "Missing Agent", "Please start/select agent.", eUserMsgOption.OK, eUserMsgSelection.None));
+            Reporter.UserMsgsPool.Add(eUserMsgKey.RestartAgent, new UserMsg(eUserMsgType.WARN, "Agent config error", "Please restart agent.", eUserMsgOption.OK, eUserMsgSelection.None));
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs]: please change message title to Agent Restart needed

File: Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs [hunk 2]
```diff
@@ -165,13 +165,19 @@ private void FillTargetAppsComboBox()
...
                 }
-                else
-                {                    
-                    Reporter.ToUser(eUserMsgKey.MissingTargetApplication, "The mapped" + mPOM.Key.ItemName + "Target Application was not found, please select new Target Application");
-                }
             }
 
-            xTargetApplicationComboBox.ItemsSource = WorkSpace.Instance.Solution.ApplicationPlatforms.Where(x => ApplicationPOMModel.PomSupportedPlatforms.Contains(x.Platform)).ToList();
+            xTargetApplicationComboBox.ItemsSource = null;
+            var targetPlatform = WorkSpace.Instance.Solution.GetTargetApplicationPlatform(mPOM.TargetApplicationKey);
```
Comment (JinendraGhodke) [inline Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs]: i think same function is called from wizard as well. Need to ensure on wizard it always shows full list

File: Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs [hunk 3]
```diff
@@ -165,13 +165,19 @@ private void FillTargetAppsComboBox()
...
+            if (targetPlatform != ePlatformType.NA)
```
Comment (JinendraGhodke) [inline Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs]: also add check if Targetplatform != null

File: Ginger/GingerCore/Drivers/JavaDriverLib/JavaDriver.cs [hunk 4]
```diff
@@ -1863,83 +1866,105 @@ List<AppWindow> IWindowExplorer.GetAppWindows()
 
         List<ElementInfo> IWindowExplorer.GetVisibleControls(List<eElementType> filteredElementType, ObservableList<ElementInfo> foundElementsList = null)
         {
+            if (filteredElementType.Count == 0 && foundElementsList == null)
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Drivers/JavaDriverLib/JavaDriver.cs]: Please do it based on AutoLearn flag

File: Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs (hunk 3 above)
Comment (ravirk91) [reply Ginger/Ginger/ApplicationModelsLib/POMModels/POMEditPage.xaml.cs]: null is not required here as function GetTargetApplicationPlatform(mPOM.TargetApplicationKey) never return null.

**Pull request merged**